### PR TITLE
64bit elliptic curve

### PR DESCRIPTION
**Pull request by psychocrypt**
- add CMake option `USE_EC_64`
- add source code define via cmake `HAVE_EC_64`
- add assempler 64bit elliptic curve implemention
  - currently used only to optimize the wallet scanning
- add test case to verify new code against old implementation

Co-authored-by: fireice-uk <fireice-uk@users.noreply.github.com>
Co-authored-by: psychocrypt <psychocryptHPC@gmail.com>

- [x] add unit test
- [x] scan wallet

### CODE REVIEW

File: src/crypto/ecops64/ecops64-c.c [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+	fe64_mul(r->z, r->x, ysubx);
+	fe64_mul(r->y, r->y, xaddy);
+	fe64_mul(r->t, t2d, p->t);
+	fe64_mul(r->x, p->z, q->z);
+	fe64_add(t0, r->x, r->x);
+	fe64_sub(r->x, r->z, r->y);
+	fe64_add(r->y, r->z, r->y);
+	fe64_sub(r->z, t0, r->t);
+	fe64_add(r->t, t0, r->t);
+	}
```
Comment (fireice-uk) [inline src/crypto/ecops64/ecops64-c.c]: Broken whitespace

Comment (psychocrypt) [reply src/crypto/ecops64/ecops64-c.c]: will update, I have not touched this file expect the license

**Comment by fireice-uk** [comment]
@psychochrypt Nice to know Monero  guys are watching the "hostile" project so closely. Hello to you too @SamsungGalaxyPlayer :)

**Comment by SamsungGalaxyPlayer** [comment]
Initial implementation of 64-bit math is a commendable step forward. I hope both projects can benefit from deeper integration for time savings.

**Comment by fireice-uk** [comment]
That's a welcome change of attitude from "everything you do is hostile". You can pull that code from next year's ryo-libre if you like it.

**Comment by SamsungGalaxyPlayer** [comment]
@fireice-uk I don't intend to have an argument here. I have full confidence that others can also use djb without license violations.

**Comment by fireice-uk** [comment]
@SamsungGalaxyPlayer The nice phase did not last very long. Oh well.

**Pull request merged**